diff --git a/src/flash/nor/nrfx.c b/src/flash/nor/nrfx.c
index d14a4ac0..4259c428 100644
--- a/src/flash/nor/nrfx.c
+++ b/src/flash/nor/nrfx.c
@@ -1417,20 +1417,6 @@ static int nrfx_handle_mass_erase_command(struct command_invocation *cmd,
 	if (res != ERROR_OK)
 		return res;
 
-	uint32_t ppfc;
-
-	res = ficr_read(chip, NRFX_FICR_PPFC, &ppfc);
-	if (res != ERROR_OK) {
-		LOG_ERROR("Couldn't read PPFC register");
-		return res;
-	}
-
-	if ((ppfc & 0xFF) == 0x00) {
-		LOG_ERROR("Code region 0 size was pre-programmed at the factory, "
-			  "mass erase command won't work.");
-		return ERROR_FAIL;
-	}
-
 	res = nrfx_erase_all(chip);
 	if (res != ERROR_OK) {
 		LOG_ERROR("Failed to erase the chip");
