diff --git a/auto.def b/auto.def
index fe6e4a2..389645f 100644
--- a/jimtcl/auto.def
+++ b/jimtcl/auto.def
@@ -306,12 +306,8 @@ if {[opt-bool references]} {
     msg-result "Enabling references"
     define JIM_REFERENCES
 }
-if {[opt-bool shared with-jim-shared]} {
-    msg-result "Building shared library"
-} else {
-    msg-result "Building static library"
-    define JIM_STATICLIB
-}
+msg-result "Building static library"
+define JIM_STATICLIB
 define VERSION [format %.2f [expr {[get-define JIM_VERSION] / 100.0}]]
 define LIBSOEXT [format [get-define SH_SOEXTVER] [get-define VERSION]]
 if {[get-define libdir] ni {/lib /usr/lib}} {
